### ISSUE TITLE
Could not start watchman; falling back to NodeWatcher for file system events.

### ISSUE THREAD

**Issue by Devansh3790** [angular/angular-cli#1390]
Hey, I am starting using angular-cli and get this error while starting my first app
Following are the details.

1. OS?  Ubuntu 14.04 LTS

2. $ ng --version:
angular-cli: 1.0.0-beta.9
node: 4.4.7
os: linux x64

Steps : 
1. ng new my-app
2. cd my-app
3. ng serve

Looks like you have a different program called watchman, falling back to NodeWatcher.
Visit http://ember-cli.com/user-guide/#watchman for more info.
File: /home/devansh/Documents/Workspace/django-angular2/my-app/src
The Broccoli Plugin: [Funnel] failed with:
Error: watch /home/devansh/Documents/Workspace/django-angular2/my-app/src ENOSPC
    at exports._errnoException (util.js:873:11)
    at FSWatcher.start (fs.js:1234:19)
    at Object.fs.watch (fs.js:1262:11)
    at NodeWatcher.watchdir (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/sane/src/node_watcher.js:144:20)
    at new NodeWatcher (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/sane/src/node_watcher.js:45:8)
    at new sane (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/sane/index.js:17:12)
    at EventEmitter.Watcher_addWatchDir [as addWatchDir] (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/broccoli-sane-watcher/index.js:93:17)
    at /home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/angular-cli-broccoli/lib/builder.js:95:35
    at lib$rsvp$$internal$$tryCatch (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:1036:16)
    at lib$rsvp$$internal$$invokeCallback (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:1048:17)
    at /home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:331:11
    at lib$rsvp$asap$$flush (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:1198:9)
    at nextTickCallbackWith0Args (node.js:420:9)
    at process._tickCallback (node.js:349:13)

The broccoli plugin was instantiated at: 
    at Funnel.Plugin (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/broccoli-plugin/index.js:10:31)
    at new Funnel (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/broccoli-funnel/index.js:44:10)
    at Angular2App._buildInputTree (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/lib/broccoli/angular2-app.js:92:7)
    at new Angular2App (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/lib/broccoli/angular2-app.js:51:41)
    at module.exports (/home/devansh/Documents/Workspace/django-angular2/my-app/angular-cli-build.js:10:10)
    at Class.module.exports.Task.extend.setupBroccoliBuilder (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/angular-cli/lib/models/builder.js:55:19)
    at Class.module.exports.Task.extend.init (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/angular-cli/lib/models/builder.js:89:10)
    at new Class (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/core-object/core-object.js:18:12)
    at Class.module.exports.Task.extend.run (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/angular-cli/lib/tasks/serve.js:15:19)
    at /home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/angular-cli/lib/commands/serve.js:64:24
    at lib$rsvp$$internal$$tryCatch (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:1036:16)
    at lib$rsvp$$internal$$invokeCallback (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:1048:17)
    at /home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:331:11
    at lib$rsvp$asap$$flush (/home/devansh/Documents/Workspace/django-angular2/my-app/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:1198:9)
    at nextTickCallbackWith0Args (node.js:420:9)
    at process._tickCallback (node.js:349:13)

Livereload server on http://localhost:49152
Serving on http://localhost:4200/

**Comment by Devansh3790** [comment]
I am searching a lot about the above issue but still not fount any solution. I found that i have that issue after my Ubuntu update. I have enabled auto update for my OS and this issue starting after update completed.

**Comment by filipesilva** [comment]
Closed as this issue was made obsolete by https://github.com/angular/angular-cli/pull/1455.

**Comment by kamok** [comment]
I just encountered this when deploying to Heroku. On development it works fine.

**Comment by tonyvince** [comment]
same problem here when trying to deploy on heroku

**Comment by kamok** [comment]
@tonyvince Nope. I gave up playing with Angular 2 before it gets out of beta. Currently building projects with Angular 1, and trying React JS.

**Comment by gnumoksha** [comment]
I had the same issue and was solved by installation of watchman https://facebook.github.io/watchman/docs/install.html

**Comment by mpavel** [comment]
I had the same issue with Watchman and I actually followed the link mentioned in the cli (http://ember-cli.com/user-guide/#watchman).
Issue resolved by installing watchman like instructed: `$ brew install watchman`

**Comment by tucq88** [comment]
I installed watchman but error still shows T_T confusing

![https://i.gyazo.com/d2a53aa3dc1552a262a027caf208db55.png](https://i.gyazo.com/d2a53aa3dc1552a262a027caf208db55.png)

**Comment by mpavel** [comment]
@tucq88, I have watchman under "/usr/local/bin/watchman"

**Comment by nevergear** [comment]
if you are using linux i think you can install watchman

`// installing watchman from source
git clone https://github.com/facebook/watchman.git
cd watchman
./autogen.sh
./configure
make
sudo make install`

**Comment by tucq88** [comment]
@mpavel 
Watchman location doesn't matter. It's because when we install watchman with `brew install watchman` , installation will use [default locations](https://facebook.github.io/watchman/docs/cli-options.html#quick-note-on-default-locations) for config. It's is `user/local/bin` which if `brew` doesn't have sudo permission, it will not be created.  You could try run `watchman` then an not found error will shows. 

So my solution is using `brew install HEAD watchman`, it similar to build it from source like @nevergear  suggestion. So `./configure` is runned, it will able to create watchman dir => Problem solved. 

TL;DR: use `brew install HEAD watchman`

**Comment by filipesilva** [comment]
https://github.com/angular/angular-cli/pull/2846 should make this warning go away. We don't actually use watchman anywhere, it's part of ember-cli. See https://github.com/angular/angular-cli/issues/2791 for more info.

**Comment by jihgao** [comment]
Try  ln -s /Users/tucp/.homebrew/bin/watchman  /usr/local/bin/watchman

**Comment by aren55555** [comment]
@tucq88 you more likely want to do `brew install --build-from-source watchman` such that you are building a release rather than directly from the latest commit (`HEAD`).

See: https://github.com/Homebrew/homebrew-core/issues/6454

**Comment by dcchristopher** [comment]
For OSX: Once you have installed watchman and can confirm by running "watchman --version", make sure it has permissions to the project folder in which you're working: "watchman -p". If this returns an error about "permission denied", you have two options: grant your current user account access to everything on your hard drive or run ng in an elevated shell, e.g. "sudo su -" (the latter worked for me). ... or just get the latest version that does away with the check and the warning ...

**Comment by tucq88** [comment]
@dcchristopher 
I got this ![image](https://i.gyazo.com/04784ec5e2e19063a23546c8e1e5f78c.png) 
Is it normal ?

**Comment by dcchristopher** [comment]
@tucq88 It looks like you're not having any permission issues related to watchman creating a state file. That's one of the first things it tries to do. For me, running "watchman" (with or without " -p") under my personal user account leads to the permissions error. If I run it in a "sudo su -" session, then it's fine (and I get the same error about insufficient arguments as you).
